performance-tests/bench_2/test_controller/main.cpp

```diff
@@ -35,7 +37,53 @@ using namespace Bench::TestController;
 
 std::string bench_root;
 
-int handle_reports(const Bench::NodeController::ReportSeq& nc_reports, const std::vector<Bench::WorkerReport>& parsed_reports, std::ostringstream& result_out);
+int handle_reports(const Bench::NodeController::ReportSeq& nc_reports,
+  const std::vector<Bench::WorkerReport>& parsed_reports,
+  const std::unordered_set<std::string>& tags,
+  std::ostringstream& result_out);
+
+void update_stats_for_tags(std::unordered_map<std::string, uint64_t>& stats,
+  const Builder::StringSeq& reported_tags,
+  const std::unordered_set<std::string>& input_tags,
+  const Builder::ConstPropertyIndex& prop)
+{
+  for (CORBA::ULong i = 0; i < reported_tags.length(); ++i) {
+    const std::string tag(reported_tags[i]);
+    if (input_tags.find(tag) != input_tags.end()) {
+      if (stats.find(tag) == stats.end()) {
+        stats[tag] = prop->value.ull_prop();
+      } else {
+        stats[tag] += prop->value.ull_prop();
+      }
+    }
+  }
+}
+
+void update_details_for_tags(std::unordered_map<std::string, std::string>& details,
+  const Builder::StringSeq& reported_tags,
+  const std::unordered_set<std::string>& input_tags,
+  const std::string& detail)
+{
+  for (CORBA::ULong i = 0; i < reported_tags.length(); ++i) {
+    const std::string tag(reported_tags[i]);
+    if (input_tags.find(tag) != input_tags.end()) {
+      details[tag] += detail;
+    }
+  }
+}
+
+void consolidate_tagged_stats(std::unordered_map<std::string, Bench::SimpleStatBlock>& stats,
```

Comment:
Is this a good candidate for the common library? Maybe once we figure out how to handle tag "expressions" or however we want to handle queries involving multiple tags?

Reply:
I don't know if there are other places that will use this function. But yes, we can move it if there are.